Replace Java String Templates; either (probably) with simple String Concatenation [or real Template Engine, but... why?]

See https://bugs.openjdk.org/browse/JDK-8329949

Noticed from https://github.com/google/google-java-format/commit/fdf4b29a6e1206f451bd96ed7967f4061921881a

This will also unblock upgrading  `google-java-formatter` from 1.21.0 to 1.24.0; that's currently blocked due to this failure:

```
java/dev/enola/web/ThingUI.java:85:50: error: did not generate token "."
            sb.append(STR."<td class=\"label\">\{link(nameValue.getKey())}</td>");
                                                 ^
tools/hello/src/main/java/dev/enola/common/hello/Hello.java:30:30: error: did not generate token "."
        return STR."hello, \{planet}";
                             ^
tools/hello/src/main/java/dev/enola/common/hello/Library.java:24:30: error: did not generate token "."
        return STR."hello, \{planet}";
                             ^
```

--- a/java/dev/enola/cli/BUILD
+++ b/java/dev/enola/cli/BUILD
@@ -27,7 +27,6 @@ java_binary(
             "SystemOutErrCapture.java",
         ],
     ),
-    jvm_flags = ["--enable-preview"],
     main_class = "dev.enola.cli.EnolaCLI",
     deps = [
         "//java/dev/enola/common",
@@ -73,7 +72,6 @@ junit_tests(
         ["*Test.java"],
         exclude = ["EnolaLoggingTest.java"],
     ),
-    jvm_flags = ["--enable-preview"],
     srcs_utils = [
         "CommandLineSubject.java",
         "SystemOutErrCapture.java",

--- a/java/dev/enola/web/BUILD
+++ b/java/dev/enola/web/BUILD
@@ -23,7 +23,6 @@ java_library(
         ["*.java"],
         exclude = ["*Test*.java"],
     ),
-    javacopts = ["--enable-preview"],
     resource_strip_prefix = "java/dev/enola/web/resources/",
     # TODO Use **/*.html instead of **/* here (like in gen/BUILD)
     resources = glob(["resources/**/*"]),
@@ -52,7 +51,6 @@ junit_tests(
     name = "tests",
     size = "medium",
     srcs = glob(["*Test.java"]),
-    jvm_flags = ["--enable-preview"],
     srcs_utils = glob(["*Abstract.java"]),
     runtime_deps = [
         "@maven//:org_slf4j_slf4j_simple",

--- a/java/dev/enola/web/ThingUI.java
+++ b/java/dev/enola/web/ThingUI.java
@@ -17,8 +17,6 @@
  */
 package dev.enola.web;
 
-import static java.lang.StringTemplate.STR;
-
 import com.google.common.escape.Escaper;
 import com.google.common.html.HtmlEscapers;
 
@@ -77,13 +75,13 @@ public class ThingUI {
     private CharSequence table(Map<String, Value> fieldsMap, String cssClass) {
         var sb = new StringBuilder("<table");
         if (!cssClass.isEmpty()) {
-            sb.append(" class=\"" + s(cssClass) + "\"");
+            sb.append(" class=\"").append(s(cssClass)).append("\"");
         }
         sb.append("><tbody>\n");
         for (var nameValue : fieldsMap.entrySet()) {
             sb.append("<tr>\n");
-            sb.append(STR."<td class=\"label\">\{link(nameValue.getKey())}</td>");
-            sb.append(STR."<td>\{value(nameValue.getValue(), "")}</td>");
+            sb.append("<td class=\"label\">").append(link(nameValue.getKey())).append("</td>");
+            sb.append("<td>").append(value(nameValue.getValue(), "")).append("</td>");
             sb.append("</tr>\n");
         }
         sb.append("</tbody></table>\n");

--- a/java/pom.xml
+++ b/java/pom.xml
@@ -30,9 +30,6 @@
         <configuration>
           <release>21</release>
           <encoding>UTF-8</encoding>
-          <compilerArgs>
-            <arg>--enable-preview</arg>
-          </compilerArgs>
         </configuration>
       </plugin>
 

--- a/tools/distro/execjar-header.bash
+++ b/tools/distro/execjar-header.bash
@@ -19,7 +19,7 @@
 # BEWARE of https://www.shellcheck.net/wiki/SC3050 (that's why Bash instead of /bin/sh)
 commandToRun="$(printf "%q " "$@")"
 if test "$commandToRun" = "'' "; then
-  eval "exec java -Xmx1G --enable-preview -jar $0"
+  eval "exec java -Xmx1G -jar $0"
 else
-  eval "exec java -Xmx1G --enable-preview -jar $0 $commandToRun"
+  eval "exec java -Xmx1G -jar $0 $commandToRun"
 fi

--- a/tools/hello/BUILD
+++ b/tools/hello/BUILD
@@ -19,8 +19,6 @@ load("@rules_java//java:defs.bzl", "java_binary", "java_library")
 java_binary(
     name = "hello",
     srcs = ["src/main/java/dev/enola/common/hello/Hello.java"],
-    javacopts = ["--enable-preview"],
-    jvm_flags = ["--enable-preview"],
     main_class = "dev.enola.common.hello.Hello",
     deps = [":hello_lib"],
 )
@@ -28,5 +26,4 @@ java_binary(
 java_library(
     name = "hello_lib",
     srcs = ["src/main/java/dev/enola/common/hello/Library.java"],
-    javacopts = ["--enable-preview"],
 )

--- a/tools/hello/src/main/java/dev/enola/common/hello/Hello.java
+++ b/tools/hello/src/main/java/dev/enola/common/hello/Hello.java
@@ -17,8 +17,6 @@
  */
 package dev.enola.common.hello;
 
-import static java.lang.StringTemplate.STR;
-
 public class Hello {
     public static void main(String[] args) {
         // System.out.println("hello, world");
@@ -27,6 +25,6 @@ public class Hello {
     }
 
     public static String helloWithJavaStringTemplate(String planet) {
-        return STR."hello, \{planet}";
+        return "hello, " + planet;
     }
 }

--- a/tools/hello/src/main/java/dev/enola/common/hello/Library.java
+++ b/tools/hello/src/main/java/dev/enola/common/hello/Library.java
@@ -17,10 +17,8 @@
  */
 package dev.enola.common.hello;
 
-import static java.lang.StringTemplate.STR;
-
 public class Library {
     public static String helloWithJavaStringTemplate(String planet) {
-        return STR."hello, \{planet}";
+        return "hello, " + planet;
     }
 }

